app/codes/p2p/sync_chain.py

```diff
@@ -29,10 +34,24 @@ def receive_block(block):
     block_index = block['block_index'] if 'block_index' in block else block['index']
     if block_index > get_last_block_index() + 1:
         sync_chain_from_peers()
+    validate_block(block)
```

Comment:
Should we add a condition here?
flag = validate_block(block)
if not flag:
    print("Block is not valid")  #can also be logger
    return False